From b72b97c4d4f41acd9869ae281314d09d68f20800 Mon Sep 17 00:00:00 2001
From: Rich Bono <rbono@redhat.com>
Date: Thu, 20 Nov 2014 21:23:35 -0500
Subject: [scsi] pm8001: Fix to remove null pointer checks that could never happen

Message-id: <1416518623-7080-15-git-send-email-rbono@redhat.com>
Patchwork-id: 100412
O-Subject: [RHEL7.1 e-stor PATCH V3 14/22] pm8001: Fix to remove null pointer checks that could never happen
Bugzilla: 1110943
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>

From: Rickard Strandqvist <rickard_strandqvist@spectrumdigital.se>

Removal of null pointer checks that could never happen

Signed-off-by: Rickard Strandqvist <rickard_strandqvist@spectrumdigital.se>
Acked-by: Suresh Thiagarajan <Suresh.Thiagarajan@pmcs.com>
Acked-by: Jack Wang <xjtuwjp@gmail.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit d98164461c885ef2655d8edceec4f8f403c0b41e)

Signed-off-by: Rich Bono <rbono@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/pm8001/pm8001_hwi.c b/drivers/scsi/pm8001/pm8001_hwi.c
index b96764e..fc3ef8e 100644
--- a/drivers/scsi/pm8001/pm8001_hwi.c
+++ b/drivers/scsi/pm8001/pm8001_hwi.c
@@ -4408,7 +4408,7 @@ static int pm8001_chip_sata_req(struct pm8001_hba_info *pm8001_ha,
 
  /* Check for read log for failed drive and return */
  if (sata_cmd.sata_fis.command == 0x2f) {
-  if (pm8001_ha_dev && ((pm8001_ha_dev->id & NCQ_READ_LOG_FLAG) ||
+  if (((pm8001_ha_dev->id & NCQ_READ_LOG_FLAG) ||
    (pm8001_ha_dev->id & NCQ_ABORT_ALL_FLAG) ||
    (pm8001_ha_dev->id & NCQ_2ND_RLE_FLAG))) {
    struct task_status_struct *ts;
-- 
1.7.1